core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1190,10 +1193,17 @@ class DAGScheduler(
     if (errorMessage.isEmpty) {
       logInfo("%s (%s) finished in %s s".format(stage, stage.name, serviceTime))
       stage.latestInfo.completionTime = Some(clock.getTimeMillis())
+
+      // Clear failure count for this stage, now that it's succeeded.
+      // We only limit consecutive failures of stage attempts,so that if a stage is
+      // re-used many times in a long-running job, unrelated failures that are spaced out in time
```

Comment:
can you delete "that are spaced out in time"?  It could be misinterpreted as there being some timeout mechanism, which there is not.